src/mdns.rs

```diff
@@ -88,6 +88,17 @@ impl From<mdns_result_t> for QueryResult {
             p = unsafe { (*p).next };
         }
 
+        #[cfg(esp_idf_version_major = "5")]
```

Comment:
`tcpip_if` -> `esp_netif` in v5.0+
see: https://github.com/espressif/esp-protocols/commit/ddc58e8220f8b4d8829cfe47cfecdf071e92c845#diff-b1f0aa635ad5c4ccef42c50cda8b8e441e9fc5b324d1ea3f7703af1ded32cb75L86-R79
maybe a better approach would be to get rid of the `Interface` enum